### PR TITLE
Made ciphers operate on streams to be usable with large files

### PR DESCRIPTION
**Pull request by TomSchalld**

### CODE REVIEW

File: src/main/java/com/somesec/crypto/encrypt/AESEncryption.java [hunk 1]
```diff
@@ -46,6 +50,27 @@ public byte[] encrypt(byte[] payload, Key key) {
...
+            random.nextBytes(nonce);
+            GCMParameterSpec spec = new GCMParameterSpec((int) resolver.getConfig(DefaultConfig.AES_GCM_TAG_LENGTH_BYTE) * ((int) resolver.getConfig(DefaultConfig.BIT_IN_A_BYTE)), nonce);
+            cipher.init(Cipher.ENCRYPT_MODE, key, spec);
+            if (plainText.available() > 0) {
+                cipherText.write(nonce);
+            }
+            while (plainText.available() > 0) {
+                cipherText.write(cipher.update(plainText.readNBytes(resolver.getConfig(DefaultConfig.MEGA_BYTE))));
+            }
+            cipherText.write(cipher.doFinal());
```
Comment (deglen) [inline src/main/java/com/somesec/crypto/encrypt/AESEncryption.java:68]: what do you think if we also call #flush() before closing the streams to ensure all last bytes are written?

File: src/test/java/com/somesec/crypto/IntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.Key;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.Security;
+
+public class IntegrationTest {
```
Comment (deglen) [inline src/test/java/com/somesec/crypto/IntegrationTest.java:30]: maybe better to rename the test suite to something related to stream encryption/decryption?

File: src/test/java/com/somesec/crypto/IntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        encryptionOperation.encrypt(Files.newInputStream(plaintextTestFile), Files.newOutputStream(ciphertextTestFile), key);
+        Assertions.assertTrue(TEN_MEGABYTE < Files.size(ciphertextTestFile));
+        final byte[] cipheredFileHash = fingerPrintFile(ciphertextTestFile);
+        final String ciphertextHash = Hex.toHexString(cipheredFileHash);
+        Assertions.assertNotEquals(plainTextHash, ciphertextHash);
+        System.out.println("Ciphertext : " + ciphertextHash);
+        final byte[] decryptedBytes = decryptionOperation.decrypt(Files.readAllBytes(ciphertextTestFile), key);
+        final String decryptedFileHash = Hex.toHexString(fingerprint(decryptedBytes));
+        Assertions.assertEquals(plainTextHash, decryptedFileHash);
+        System.out.println("Decryptedtext : " + decryptedFileHash);
```
Comment (deglen) [inline src/test/java/com/somesec/crypto/IntegrationTest.java:79]: system out?

File: src/main/java/com/somesec/crypto/encrypt/AESEncryption.java (hunk 1 above)
Comment (TomSchalld) [reply src/main/java/com/somesec/crypto/encrypt/AESEncryption.java:68]: Yes thats definitely a good idea.

File: src/test/java/com/somesec/crypto/IntegrationTest.java (hunk 3 above)
Comment (TomSchalld) [reply src/test/java/com/somesec/crypto/IntegrationTest.java:79]: Embarrassing

**Pull request merged**